test/types/community/cursor.test-d.ts

```diff
@@ -98,3 +98,42 @@ void async function () {
     expectType<number>(item.foo);
   }
 };
+
+interface InternalMeme {
+  _id: string;
+  owner: string;
+  receiver: string;
+  createdAt: Date;
+  expiredAt: Date;
+  description: string;
+  likes: string;
+  private: string;
+  replyTo: string;
+  imageUrl: string;
+}
+
+interface PublicMeme {
+  myId: string;
+  owner: string;
+  likes: number;
+  someRandomProp: boolean; // Projection makes no enforcement on anything
+  // the convenience parameter project<X>() allows you to define a return type,
+  // otherwise projections returns a generic Document
+}
+
+const publicMemeProjection = {
+  myId: { $toString: '$_id' },
+  owner: { $toString: '$owner' },
+  receiver: { $toString: '$receiver' },
+  likes: '$totalLikes' // <== cause of TS2345 error
+};
+
+expectType<PublicMeme[]>(
+  await new Db(new MongoClient(''), '')
+    .collection<InternalMeme>('memes')
+    .find({ _id: { $in: [] } })
+    .sort({ _id: -1 })
+    .limit(3)
+    .project<PublicMeme>(publicMemeProjection) // <== location of TS2345 error
```

Comment:
what's TS2345?

Reply:
Copy pasta-ed from the ticket its the type error that was occuring with the projection constraints, I put in the actual message for clarity.

Reply:
I think including the ticket number would make sense, to provide context in the future about which TS2345 error this is referring to.